src/UI/SelectList.elm

```diff
@@ -0,0 +1,232 @@
+module UI.SelectList exposing (SelectList, selectList, toEl, withActionBar, withOptions, withSearchField, withSelected)
+
+import Element exposing (Attribute, Element, fill, shrink)
+import Element.Background as Background
+import Element.Border as Border
+import Element.Events as Events
+import Element.Font as Font
+import UI.ActionBar as ActionBar
+import UI.Icon as Icon exposing (Icon)
+import UI.Internal.Palette as Palette
+import UI.RenderConfig exposing (RenderConfig)
+import UI.Text as Text
+import UI.TextField as TextField
+import UI.Utils.ARIA as ARIA
+import UI.Utils.Element as Element
+
+
+type alias SearchOptions object msg =
+    { label : String
+    , searchMsg : String -> msg
+    , currentFilter : Maybe ( String, String -> object -> Bool )
+    }
+
+
+type alias Options object msg =
+    -- FUTURE: sort: Maybe (List object -> List object)
+    { itens : List object
+    , searchField : Maybe (SearchOptions object msg)
+    , actionBar : Maybe ( String, String -> Icon, msg ) -- TODO
+    , isSelected : Maybe (object -> Bool)
```

Comment:
I would say that in the middle of the code when I saw `isSelected` I thought it was a `Bool` instead of a function that returns a Bool, so I would suggest calling maybe `isSelectedFn` or something like that because normally we would use `is*` as a Bool, but it's up to you.

Reply:
I prefer `selected` as a boolean, `isSelected` to `sth -> Bool`. We should start writing contribution guidelines...